scripts/zones/Lower_Jeuno/npcs/Zalsuhm.lua

```diff
@@ -3,114 +3,110 @@
-
-    if (unlockingAMyth == QUEST_AVAILABLE) then
-        local zalsuhmUpset = player:getVar("Upset_Zalsuhm");
-        if (player:needToZone() and zalsuhmUpset > 0) then -- Zalsuhm is still angry
-            player:startEvent(10090);
+    if unlockingAMyth == QUEST_AVAILABLE then
+        if player:needToZone() and player:getVar("Upset_Zalsuhm") > 0 then
+            player:startEvent(10090)
         else
-            if (zalsuhmUpset > 0) then
-                player:setVar("Upset_Zalsuhm", 0);
+            if player:getVar("Upset_Zalsuhm") > 0 then
+                player:setVar("Upset_Zalsuhm", 0)
             end
 
-            if (nyzulWeapon) then -- The player has a Nyzul weapon in the mainHand, try to initiate quest
-                player:startEvent(10086, mainJobId);
+            if nyzulWeaponMain or nyzulWeaponRanged then
+                player:startEvent(10086, mainJobId)
             else
-                player:startEvent(10085); -- Default dialog
+                player:startEvent(10085)
             end
         end
-    elseif (unlockingAMyth == QUEST_ACCEPTED) then -- Quest is active for current job
-        player:startEvent(10087); -- Zalsuhm asks for the player to show him the weapon if they sense a change
-    else -- Quest is complete for the current job
-        player:startEvent(10089);
+    elseif unlockingAMyth == QUEST_ACCEPTED then
+        player:startEvent(10087)
+    else
+        player:startEvent(10089)
     end
-    
-end;
+end
 
 function onEventUpdate(player,csid,option)
-
-    -- printf("LowerJeuno_Zalsuhm.onEventUpdate() - ");
-    -- printf("\tCSID: %u", csid);
-    -- printf("\tRESULT: %u", option);
-
-end;
+end
 
 function onEventFinish(player,csid,option)
-
-    -- printf("LowerJeuno_Zalsuhm.onEventFinish() - ");
-    -- printf("\tCSID: %u", csid);
-    -- printf("\tRESULT: %u", option);
-
-    -- Zalsuhm wants to research the player's Nyzul Weapon
-    if (csid == 10086) then
-        -- The player chose "He has shifty eyes" (turns down the quest)
-        if (option == 53) then
-            player:setVar("Upset_Zalsuhm", 1);
-            player:needToZone(true);
-        elseif (option <= dsp.job.SCH) then -- Just to make sure we didn't get into an invalid state
-            -- The player chose "More power" (accepts the quest)
-            local questId = getQuestId(option);
-            player:addQuest(JEUNO, questId);
+    local questId = getQuestId(option)
+    if csid == 10086 then
+        if option == 53 then
+            player:setVar("Upset_Zalsuhm", 1)
+            player:needToZone(true)
+        elseif option <= dsp.job.SCH then
+            player:addQuest(JEUNO, questId)
         end
-    elseif (csid == 10088 and option <= dsp.job.SCH) then -- The quest is completed
-        local questId = getQuestId(option);
-        player:completeQuest(JEUNO, questId);
+    elseif csid == 10088 and option <= dsp.job.SCH then
+        local jobs = 
```

Comment:
this should really be called like, weaponskills.

There's also a global/weaponskillids.lua that you can set these values to for readability instead of "magic numbers."

Reply:
these are not the weapon skill ids these are the unlock_ids they are not defined unless you want me to add them to the weaponskillids.lua? dsp.ws_unlock.KINGSJUSTICE ?

Reply:
I'm not trying to pile this onto you @KnowOne134, these are just general questions for everyone:

Does DSP not have lua enums for weaponskills yet?

Should they be added? Should this PR be using them?